packages/manager/apps/pci-savings-plan/src/utils/formatter/date.ts

```diff
@@ -10,3 +10,18 @@ export const toLocalDateUTC = (date: string, locale: string) =>
     minute: '2-digit',
     timeZoneName: 'short',
   });
+
+export const getLastTwelveMonths = (locale: string) => {
+  const months = [];
+  const date = new Date();
+  for (let i = 0; i < 12; i += 1) {
+    months.push(
+      new Intl.DateTimeFormat(locale.replace('_', '-'), {
+        month: 'long',
+        year: 'numeric',
+      }).format(date),
+    );
+    date.setMonth(date.getMonth() - 1);
+  }
+  return months;
+};
```

Comment:
If you want a full period format I would recommend using https://date-fns.org/v4.1.0/docs/eachMonthOfInterval with the format from date-fns.

If you only want the months formatted I would suggest initializing the DateTimeFormat in the beginning, create an array, and fill it with the formatted months